### PR TITLE
Remember page state

### PR DESCRIPTION
**Pull request by krystian8207**
Previously all the pages were initially hidden and shown on bookmark change.
This solution have an important disadvantage:
- hidden page inputs are not bind,
- you need to use `suspendWhenHidden = FALSE` for all the inputs in order to update them.
We decided to hide all the pages with `visibility: hidden`.
This forces all the pages to be rendered during runtime, but between page state is remembered and not rendered again.

What's more the API is more consistent now. You should use `router$ui` instead of `router_ui()` and `router$server(input, output, session)` instead of `router(input, output, session)`.

### CODE REVIEW

File: R/router.R [hunk 1]
```diff
@@ -1,11 +1,28 @@
 ROUTER_UI_ID <- '_router_ui'
 
+attach_attribs <- function(ui, path) {
```
Comment (dokato) [inline R/router.R:3]: short documentation needed

File: R/router.R [hunk 2]
```diff
@@ -119,18 +137,42 @@ create_router_callback <- function(root, routes) {
...
-      shiny::insertUI(
-        "#router-page-wrapper", "afterBegin",
-        shiny::div(id = "router-page-content", routes[[page_path]][["ui"]])
-      )
+      session$sendCustomMessage("switch-ui", page_path)
     })
   }
 }
 
+router_server <- function(router) {
```
Comment (dokato) [inline R/router.R:148]: should this be exported, or users should use only `make_router`?

File: R/router.R [hunk 3]
```diff
@@ -119,18 +137,42 @@ create_router_callback <- function(root, routes) {
...
+  create_router_callback(router$root, router$routes)
+}
+
+router_ui <- function(router) {
```
Comment (dokato) [inline R/router.R:152]: docs please

**Review by dokato — CHANGES_REQUESTED** [review]
I really like these changes. They solve a lot of issues. Please address outstanding minor comments.

Also, from my tests on internal server, looks like it solves #68. I was thinking of introducing backward compatibility with router and router_ui calls, but I can see that this would be difficult in the current setting, so maybe we can ignore.

File: R/router.R (hunk 2 above)
Comment (krystian8207) [reply R/router.R:154]: This is internal function, and users should use only make_router.

File: R/router.R (hunk 3 above)
Comment (krystian8207) [reply R/router.R:165]: I'll add this but this is internal function as well.

File: R/router.R (hunk 2 above)
Comment (dokato) [reply R/router.R:154]: ok, but internal function should have short docs too, please

**Review by dokato — APPROVED** [review]
I tested locally and on R Connect, looks good now 👍

**Pull request merged**